### ISSUE TITLE
Build Error: cannot use a string pattern on a bytes-like object

### ISSUE THREAD

**Issue by GarrettKucinski** [libxmljs/libxmljs#469]
Getting an error on falling-back-to-build

```
> libxmljs@0.18.4 install /Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http

node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.4 and node@8.0.0 (node-v57 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
```

Any thoughts on this? Not sure what do do about it?
```
In file included from ../vendor/libxml/xpath.c:476:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity' [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
2 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
Traceback (most recent call last):
  File "./gyp-mac-tool", line 611, in <module>
    sys.exit(main(sys.argv[1:]))
  File "./gyp-mac-tool", line 28, in main
    exit_code = executor.Dispatch(args)
  File "./gyp-mac-tool", line 43, in Dispatch
    return getattr(self, method)(*args[1:])
  File "./gyp-mac-tool", line 246, in ExecFilterLibtool
    if not libtool_re.match(line) and not libtool_re5.match(line):
TypeError: cannot use a string pattern on a bytes-like object
make: *** [Release/xml.a] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:285:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Darwin 16.6.0
gyp ERR! command "/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node" "/Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--loglevel=http" "--module=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release/xmljs.node" "--module_name=xmljs" "--module_path=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release"
gyp ERR! cwd /Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs
gyp ERR! node -v v8.0.0
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node /Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release/xmljs.node --module_name=xmljs --module_path=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:125:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:887:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
node-pre-gyp ERR! System Darwin 16.6.0
node-pre-gyp ERR! command "/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node" "/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
node-pre-gyp ERR! cwd /Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs
node-pre-gyp ERR! node -v v8.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.36
node-pre-gyp ERR! not ok
Failed to execute '/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node /Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release/xmljs.node --module_name=xmljs --module_path=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! libxmljs@0.18.4 install: `node-pre-gyp install --fallback-to-build --loglevel http`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the libxmljs@0.18.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
```

**Comment by jakeandreoli** [comment]
Same here.

**Comment by rchipka** [comment]
This seems to be a node-gyp issue.

Try to install using these commands:

 ```
npm config set python /path/to/executable/python2.7
npm install --python=python2.7
```

Yet another reason why this library needs pre-compiled binary support for Mac OS.

**Comment by cminardi** [comment]
I got this error when building another project that was depending on `node-gyp`.

The solution above did not work for me (I am on MacOS Sierra 10.12.4) - the error kept popping up.
What worked instead was to "hide" the fact that I have Python 3 at all by taking it out of the `PATH` definition `~/.bash_profile`
```
PATH="/Library/Frameworks/Python.framework/Versions/2.7/bin:${PATH}"
export PATH

#PATH="/Library/Frameworks/Python.framework/Versions/3.6/bin:${PATH}"
#export PATH
```
So now when I run `python` it only finds the 2.7 version, and `npm install` completes successfully.

**Comment by tuananh** [comment]
@cminardi isn't that less convenient than the above method by @rchipka ?

**Comment by cminardi** [comment]
Yep, it's definitely less convenient - and not desirable at all. But as I said, their suggestion didn't work for me somehow, I thought I could share in case someone else finds themselves in this situation.

**Comment by tuananh** [comment]
@cminardi i see. i missed the part where you said it wasn't working for you.

**Comment by rchipka** [comment]
Precompiled binary support coming soon (see: #470).

**Comment by rchipka** [comment]
Fixed in v0.18.7

**Comment by jordanmkoncz** [comment]
Just want to suggest an alternative solution for anyone who has this issue. If you have Python installed using `pyenv` (https://github.com/pyenv/pyenv), this is very easy to do (and if you don't have Python installed using `pyenv`, I highly recommend it. Basic steps with `pyenv`:

- `pyenv install 2.7.16` (assuming a version of Python 2 is not already installed)
- `pyenv shell 2.7.16`
- `npm install`

These steps will set your current shell/terminal so that only Python 2 is available, and then when you run `npm install` from that same shell, the installation/build works.
